pandas/core/arrays/period.py

```diff
@@ -389,6 +389,11 @@ def freqstr(self) -> str:
     def __array__(
         self, dtype: NpDtype | None = None, copy: bool | None = None
     ) -> np.ndarray:
+        if copy is False:
+            raise ValueError(
+                "Unable to avoid copy while creating an array as requested."
+            )
```

Comment:
Strictly speaking, the 

```
        if dtype == "i8":
            return self.asi8
```

just below is zero copy, so a `copy=False` is possible in that case

Reply:
Ah, and more importantly, that means that that branch is not honoring `copy=True`

Reply:
Will look into this and other fixups tomorrow.  Right, wrong `copy=True` is a problem, I definitely err'd on the error side for `copy=False`, so let me change it here (as well).